virtcontainers/clh.go

```diff
@@ -410,6 +412,33 @@ func (clh *cloudHypervisor) getThreadIDs() (vcpuThreadIDs, error) {
 	return vcpuInfo, nil
 }
 
+func (clh *cloudHypervisor) hotplugBlockDevice(drive *config.BlockDrive) error {
+	cl := clh.client()
+	ctx, cancel := context.WithTimeout(context.Background(), clhHotPlugAPITimeout*time.Second)
+	defer cancel()
+
+	_, _, err := cl.VmmPingGet(ctx)
+	if err != nil {
+		return openAPIClientError(err)
+	}
+
+	if drive.Pmem {
+		err = fmt.Errorf("pmem device hotplug not supported")
+	} else {
+		blkDevice := chclient.DiskConfig{
+			Path:      drive.File,
+			Readonly:  drive.ReadOnly,
```

Comment:
Why ReadOnly?

Reply:
good catch, @likebreath  I wonder if this could be cached by the CI if we enable a job for block device

Reply:
@amshinde @jcvenegas I assume it would make sense to respect the `Readonly` field from the input `drive` struct. It is a good point that whether it is a problem that the `drive.Readonly` is set to be `true`. If it is a problem and should never be expected, I can add a check on `drive.Readonly` and report error if it is `true`.

Reply:
Thanks for clarifying I see is the value of the drive for a moment I got confused, I was thinking that was literally `ReadOnly`